src/Chapter3.hs

```diff
@@ -1006,6 +1190,122 @@ Implement data types and typeclasses, describing such a battle between two
 contestants, and write a function that decides the outcome of a fight!
 -}
 
+class Fighter a where
+  getAttack :: a -> Attack
+  getHealth :: a -> Health
+  getDefense :: a -> Defense
+  getName :: a -> String
+  updateHealth :: a -> Health -> a
+  takeAction :: Fighter b => a -> b -> String
```

Comment:
Yeah, that's a correct solution of using the `Figher` constraint here. So, when implementing an instance, you know all implementation details of the type `a`, with `b` you can work only using methods of the `Fighter` typeclass